### PR TITLE
[Bugfix] Make iphone info header fadeout when participant list is displayed

### PR DESCRIPTION
**Pull request by jimchou-dev**
## Purpose

* Bugfix for info header not disappear when participant list drawer is displayed
* For iPad the info header remains displayed due to fluent UI drawer need a reference point

## Does this introduce a breaking change?

```
[ ] Yes
[ x ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ x ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What to Check
Verify that the following are valid
* Check when participant list drawer is displayed for iPhone the info header will fade out and disappear
* Check when participant list drawer is displayed for iPad the info header will remain displayed until the drawer is dismissed by tapping background the info header will also be dismissed

### CODE REVIEW

File: AzureCommunicationUI/AzureCommunicationUI/Presentation/SwiftUI/Calling/CallingViewComponent/InfoHeaderViewModel.swift [hunk 1]
```diff
@@ -38,7 +38,9 @@ class InfoHeaderViewModel: ObservableObject {
 
     func showParticipantListButtonButtonTapped() {
         logger.debug("Show participant list button tapped")
-        self.infoHeaderDismissTimer?.invalidate()
+        if UIDevice.current.userInterfaceIdiom == .pad {
```
Comment (palatter) [inline AzureCommunicationUI/AzureCommunicationUI/Presentation/SwiftUI/Calling/CallingViewComponent/InfoHeaderViewModel.swift:41]: Should we be querying device directly in ViewModel?

File: AzureCommunicationUI/AzureCommunicationUI/Presentation/SwiftUI/Calling/CallingViewComponent/InfoHeaderView.swift [hunk 2]
```diff
@@ -16,30 +16,32 @@ struct InfoHeaderView: View {
...
-                    .padding(EdgeInsets(top: infoLabelHorizontalPadding,
-                                        leading: 0,
-                                        bottom: infoLabelHorizontalPadding,
-                                        trailing: 0))
-                    .foregroundColor(foregroundColor)
-                    .font(Fonts.caption1.font)
-                Spacer()
-                participantListButton
+        ZStack {
+            if viewModel.isInfoHeaderDisplayed {
```
Comment (AmyL219) [inline AzureCommunicationUI/AzureCommunicationUI/Presentation/SwiftUI/Calling/CallingViewComponent/InfoHeaderView.swift:20]: Could we abstract this part of the code into properties of `some View`, like infoContentView, for better readability

**Review by AmyL219 — APPROVED** [review]
LGTM on code

**Review by palatter — APPROVED** [review]
Looks good.

**Pull request merged**